src/snappy_device_agents/devices/multi/__init__.py

```diff
@@ -89,9 +90,15 @@ def runtest(self, args):
         snappy_device_agents.logmsg(logging.INFO, "END testrun")
         return exitcode
 
-    def get_job_list_data(self):
-        """Read job_list.json and return the data"""
-        with open("job_list.json") as job_list_file:
+    def get_job_list_data(self, job_list_file: str = "job_list.json") -> list:
+        """Read job_list.json and return the list data"""
+        if not os.path.exists(job_list_file):
+            logmsg(
+                logging.ERROR,
+                "Unable to find multi-job data file, job_list.json not found",
+            )
+            return []
+        with open(job_list_file) as job_list_file:
```

Comment:
idk what context this runs in. But if there is a possibility of race conditions then this could still fail. Alternatively, we could wrap it in try/except and catch `FileNotFoundError`

Reply:
Don't we check it in the if statement above?
```python
        if not os.path.exists(job_list_file):
            logmsg(
                logging.ERROR,
                "Unable to find multi-job data file, job_list.json not found",
            )
            return []
```

Reply:
There are two places this gets called:
1. In the test run phase of the multi-device agent. In order to get here, it needs to have finished the provisioning phase first, which is where this file is exported so that we can re-read it in the next phase (this are two separate invocations of the same tool, so not a concern for a race).
2. In the cleanup phase, which we call when the entire run is complete before marking the job complete

I don't see either of these as a concern for a race. The only (funny) possibility I can imagine is if you decided to construct a job with no provisioning phase. This would be pointless for a multi-device job, but technically possible. This would print an error once it gets to the test run phase, and you could still technically use it to run something in the agent host context if that's really what you wanted to do. But you wouldn't have DEVICE_IP info for the downstream DUTs because... well, you didn't tell it to provision any! :)

I also considered using FileNotFoundError. I couldn't see any big advantage either way, just personal preference. I'm happy to change it if you can think of a good reason to use the exception instead.

Reply:
Then if race conditions aren't an issue then this should be fine